tests/engine/test_process.py

```diff
@@ -497,3 +497,46 @@ def test_not_required_accepts_none():
     process = instantiate_process(runner, TestNotRequiredNoneProcess, valid_type=orm.Int(1), any_type=orm.Bool(True))
     assert process.inputs.valid_type == orm.Int(1)
     assert process.inputs.any_type == orm.Bool(True)
+
+
+class TestMetadataInputsProcess(Process):
+    """Process with various ports that are ``metadata=True``."""
+
+    _node_class = orm.WorkflowNode
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+        spec.input('metadata_port', metadata=True)
+        spec.input('metadata_port_non_serializable', metadata=True)
+        spec.input_namespace('metadata_portnamspace', metadata=True)
```

Comment:
I guess for the test this did not matter as the correct namespace was created on the fly?
```suggestion
        spec.input_namespace('metadata_portnamespace', metadata=True)
```